locallib.php

```diff
@@ -575,6 +575,26 @@ function attendance_random_string($length=6) {
     return $string;
 }
 
+/**
+ * This functions checks if this session is open for students.
+ *
+ * @param stdclass $sess the session record from attendance_sessions.
+ * @return boolean
+ */
+function attendance_session_open_for_students($sess) {
```

Comment:
I have fixed this. 

I wanted to highlight the fact that this is dependent on studentscanmarksessiontime  setting which is enabled by default, it stops a user from submitting attendance outside the duration of a session. 
So in the example of a lecture starting at 9am and the teacher has set "studentsearlyopentime" to 5min, at 8:54am the link should say "submit future absence", at 8:55am it should show "submit attendance". This scenario will be possible if studentscanmarksessiontime is not checked as otherwise at 8:54am the link to submit will just not be visible.